src/main/java/io/vertx/kafka/client/common/KafkaClientOptions.java

```diff
@@ -70,8 +107,52 @@ public KafkaClientOptions setConfig(String key, Object value) {
     return this;
   }
 
+  /**
+   * @return the kafka tracing policy
+   */
+  public TracingPolicy getTracingPolicy() {
+    return tracingPolicy;
+  }
+
+  /**
+   * Set the Kafka tracing policy.
+   *
+   * @param tracingPolicy the tracing policy
+   * @return a reference to this, so the API can be used fluently
+   */
+  public KafkaClientOptions setTracingPolicy(TracingPolicy tracingPolicy) {
+    this.tracingPolicy = tracingPolicy;
+    return this;
+  }
+
+  /**
+   * @return the Kafka "peer address" to show in trace tags
+   */
+  public String getTracePeerAddress() {
+    // Search for peer address in config if not provided
+    if (tracePeerAddress == null) {
+      if (config != null) {
```

Comment:
we should have this logic not here but instead when building the ConsumerTracer. Otherwise we will have the sequence : options.setTracePeerAddress(null), then options.getTracePeerAddress() that might return not null which does not make sens.

Reply:
done, PR updated

Reply:
uh, wait a minute :-)

Reply:
done